pkg/bmh/list.go

```diff
@@ -0,0 +1,107 @@
+package bmh
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	bmhv1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+)
+
+// List returns bareMetalHosts inventory in the given namespace.
+func List(apiClient *clients.Settings, nsname string) ([]*BmhBuilder, error) {
```

Comment:
Please add an goclient.ListOptions argument so that users can be more granular regarding the BMHs that they want to list